openhands/integrations/bitbucket/bitbucket_service.py

```diff
@@ -0,0 +1,374 @@
+
+                repositories.append(
+                    Repository(
+                        id=id_value,
+                        full_name=f'{repo.get("workspace", {}).get("slug", "")}/{repo.get("slug", "")}',
+                        git_provider=ProviderType.BITBUCKET,
+                        is_public=repo.get('is_private', True) is False,
+                        stargazers_count=None,  # Bitbucket doesn't have stars
+                        link_header=headers.get('Link', ''),
+                        pushed_at=repo.get('updated_on'),
+                    )
+                )
+        except Exception as e:
+            logger.warning(f'Failed to get public repositories from Bitbucket: {e}')
+
+        return repositories
+
+    async def _get_private_user_repositories(self, sort: str) -> list[Repository]:
+        """Get private repositories for the authenticated user using workspaces endpoint."""
+        repositories = []
+
+        try:
+            # Get user's workspaces
+            workspaces_url = f'{self.BASE_URL}/workspaces'
+            workspaces_data, _ = await self._make_request(workspaces_url)
+
+            for workspace in workspaces_data.get('values', []):
+                workspace_slug = workspace.get('slug')
+                if not workspace_slug:
+                    continue
+
+                # Get repositories for this workspace
+                workspace_repos_url = f'{self.BASE_URL}/repositories/{workspace_slug}'
+                repos_data, headers = await self._make_request(workspace_repos_url)
+
+                for repo in repos_data.get('values', []):
+                    # Convert UUID to an integer hash for compatibility with Repository model
+                    uuid = repo.get('uuid', '')
+                    # Remove curly braces if present and use hash of the UUID string as the ID
+                    if uuid.startswith('{') and uuid.endswith('}'):
+                        uuid = uuid[1:-1]
+                    id_value = hash(uuid) % (
+                        2**31
+                    )  # Ensure it's a positive 32-bit integer
+
+                    repositories.append(
+                        Repository(
+                            id=id_value,
+                            full_name=f'{repo.get("workspace", {}).get("slug", "")}/{repo.get("slug", "")}',
+                            git_provider=ProviderType.BITBUCKET,
+                            is_public=repo.get('is_private', True) is False,
+                            stargazers_count=None,  # Bitbucket doesn't have stars
+                            link_header=headers.get('Link', ''),
+                            pushed_at=repo.get('updated_on'),
+                        )
+                    )
+        except Exception as e:
+            logger.warning(f'Failed to get private repositories from Bitbucket: {e}')
+
+        return repositories
+
+    async def get_repositories(self, sort: str, app_mode: AppMode) -> list[Repository]:
```

Comment:
@openhands search the bitbucket documentation for the endpoints access when calling `get_repositories`

Do the repos in get_public_user_repositories overlap with the repos in get_private_user_repositories? if so then remove the implementation for `get_public_user_repositories` and move the code from `get_private_user_repositories` into `get_repositories`